flask_smorest/spec/__init__.py

```diff
@@ -131,25 +141,34 @@ def _openapi_json(self):
     def _openapi_redoc(self):
         """Expose OpenAPI spec with ReDoc"""
         return flask.render_template(
-            "redoc.html", title=self.spec.title, redoc_url=self._redoc_url
+            "redoc.html",
+            spec_url=flask.url_for(f"{self._get_doc_blueprint_name()}.openapi_json"),
+            title=self.spec.title,
+            redoc_url=self._redoc_url,
         )
 
     def _openapi_swagger_ui(self):
         """Expose OpenAPI spec with Swagger UI"""
         return flask.render_template(
             "swagger_ui.html",
             title=self.spec.title,
+            spec_url=flask.url_for(f"{self._get_doc_blueprint_name()}.openapi_json"),
```

Comment:
would those name always be url safe?

Reply:
The only restriction to a blueprint's name is that period (`'.'`) is not allowed. And `config_prefix` will not work with period either.